lib/ansible/modules/cloud/azure/azure_rm_subnet.py

```diff
@@ -228,6 +253,10 @@ def exec_module(self, **kwargs):
                         changed = True
                         results['network_security_group']['id'] = nsg.get('id')
                         results['network_security_group']['name'] = nsg.get('name')
+                if self.route_table != results['route_table'].get('id'):
+                    changed = True
+                    results['route_table']['id'] = self.route_table
+                    self.log("CHANGED: subnet {0} route_table".format(self.name))
```

Comment:
Maybe: `"CHANGED: subnet {0} route_table to {1}".format(self.name, route_table['name'])` (or similar) is less ambiguous